docs/fastdds/xtypes/language_binding.rst

```diff
@@ -0,0 +1,827 @@
+            :language: idl
+            :start-after: //!--IDL_MAPS
+            :end-before: //!--
+
+    .. tab:: XML
+
+        .. literalinclude:: /../code/XMLTester.xml
+            :language: xml
+            :start-after: <!-->XML_MAPS<-->
+            :end-before: <!--><-->
+
+    .. tab:: C++
+
+        .. literalinclude:: /../code/DDSCodeTester.cpp
+            :language: c++
+            :start-after: //!--CPP_MAPS
+            :end-before: //!--
+
+For a detailed explanation about the XML definition of this type,
+please refer to :ref:`Maps <xmldynamictypes_map>`.
+
+Aggregated Types
+""""""""""""""""
+
+.. _xtypes_supportedtypes_structure:
+
+Structures
+**********
+
+Structures are the most common complex types, they allow to add any kind of members inside them.
+They do not have any value, they are only used to contain other types.
+
+The function |DynamicTypeBuilderFactory::create_type| is used to create a new structure type.
+The ``TypeKind`` used to identify Structures is ``TK_STRUCTURE``.
+
+To manage the types inside the structure, users can call the :func:`get` and :func:`set` functions according to the 
+kind of the type inside the structure using their ``ids``.
+If the structure contains a complex value, |DynamicData::loan_value| should be used to access it and 
+|DynamicData::return_loaned_value| to release that loan.
+|DynamicData-api| manages the loaned values and users can not loan a value that has been loaned previously 
+without calling |DynamicData::return_loaned_value| before.
+
+
+.. tabs::
+
+    .. tab:: IDL
+
+        .. literalinclude:: /../code/DynamicTypesIDLExamples.idl
+            :language: idl
+            :start-after: //!--IDL_STRUCT
+            :end-before: //!--
+
+    .. tab:: XML
+
+        .. literalinclude:: /../code/XMLTester.xml
+            :language: xml
+            :start-after: <!-->XML_STRUCT<-->
+            :end-before: <!--><-->
+
+    .. tab:: C++
+
+        .. literalinclude:: /../code/DDSCodeTester.cpp
+            :language: c++
+            :start-after: //!--CPP_STRUCT
+            :end-before: //!--
+
+For a detailed explanation about the XML definition of this type,
+please refer to :ref:`Struct <xmldynamictypes_struct>`.
+
+.. _xtypes_supportedtypes_union:
+
+Unions
+******
+
+Unions are a special kind of structures where only one of the members is active at the same time.
+The ``TypeKind`` used to identify Unions is ``TK_UNION``.
+
+To control these members, users must set the ``discriminator`` type that is going to be used to select the current 
+member calling the |TypeDescriptor::discriminator_type| function of the union type.
+The ``discriminator`` itself is a DynamicType of any primitive type, string type or union type.
+
+Additionally users can use the function |MemberDescriptor::label| with one or more values to set the labels of each 
+member of the union, and |MemberDescriptor::is_default_label| to specify the default label of the union.
```

Comment:
The discriminator of a union must be of one of the following types:
•Boolean
•Byte
•Char8, Char16
•Int8, UInt8, Int16, UInt16, Int32, UInt32, Int64, UInt64
•Any enumerated type
•Any alias type that resolves, directly or indirectly, to one of the aforementioned types.

Reply:
added in [e38d0d8](https://github.com/eProsima/Fast-DDS-docs/pull/730/commits/e38d0d856970dc4cdf6d7a04f3490344f7c903b1)